### ISSUE TITLE
[Doc]: Documentation should clearly describe how a user can make matplotlib load custom fonts per default

### ISSUE THREAD

**Issue by benmaier** [matplotlib/matplotlib#24872]
### Documentation Link

https://matplotlib.org/stable/users/explain/fonts.html#how-matplotlib-selects-fonts

### Problem

The documentation is vague regarding how one should go about letting matplotlib find a directory filled with custom font files.

My current problem is that I want matplotlib, per default, to load a few font files that I cannot install systemwide, because I do not have the rights to do so (linux server). Typically, on Linux, custom fonts are stored either in `~/.fonts/` or `~/.local/share/fonts/`. Yet, even after rebuilding the linux font cache, matplotlib does not find any of the custom fonts. Previously, I made that happen by copy-pasting the files to matplotlib's `.../mpl-data/ttf/` directory, but that does not seem to work anymore (matplotlib version: 3.6.2). In earlier version there existed a `fontList.cache` but that does not exist anymore either. 

### Suggested improvement

There should be an extra section explaining how to set up a custom font-file or font-directory that matplotlib will remember to search when imported. Obviously I'm fine with whatever solution any current matplotlib version has for this problem, but the documentation is lacking clear communication on how to solve it.

**Comment by benmaier** [comment]
I don't think so. If I understood #24419 and #24866 correctly, the proposed solution uses `font_manager.addFont` to add a font file to the manager. I've tried this and the fonts are only added for this particular session, not permanently. Which means that in every script/project I use matplotlib in, I manually have to add the fonts to the font manager using the proposed piece of code. It's not a very sustainable solution. In case there actually is no sustainable solution, I'm happy to open an issue with a feature request. I'm also happy to stand corrected in case I understood sth wrong :)

**Comment by tacaswell** [comment]
The cache file that matters is now `fontlist-v330.json` and https://github.com/matplotlib/matplotlib/blob/4278c8a5a33ba1d6911ce4a095efe53af985efd0/lib/matplotlib/font_manager.py#L144-L158 is where we build the directories to check for fonts.  What you are doing _should_ work.

**Comment by benmaier** [comment]
Curious. Where would this json-file appear? and how do I trigger to rebuild? In earlier versions there was `font_manager.rebuild()`.

**Comment by oscargus** [comment]
It should reside in `~/.matplotlib` or similar.

**Comment by benmaier** [comment]
Ok, solved it. The `fontlist-v330.json` was in `~/.cache/matplotlib/`.  I deleted it, then it worked. The env `$MPLCONFIGDIR` was empty, though. 

I'd say my initial point stands, though: the documentation does not do a particularly great job at communicating how to solve this particular problem. Since many encountered the same problem in the past, Google is filled with solutions on how to achieve loading custom fonts in older versions of matplotlib, which, in my opinion, makes it all the more important that the issue is addressed in the official documentation for current versions.

**Comment by tacaswell** [comment]
I think we need a public "please nuke the font cache and rebuild it" function in `matplotlib.font_manager`.  That would I think solve a lot of these problems (and alleviate the need for users to understand the details of how we cache the locations and meta-data of the fonts).

**Comment by jklymak** [comment]
I don't think it would hurt to document the cache file location as well, and/or provide a public function to expose it. Even if we don't want to support people editing this manually, it may help folks debug their font issues if they can see what is going on.

**Comment by tfpf** [comment]
I think the cache file location is returned by `matplotlib.get_cachedir()`. I remember seeing it on the website.

**Comment by vrx-** [comment]
Hi. I came across this issue while trying to install a [dyslexic-friendly font](https://opendyslexic.org/) in my matplotlib. I was following a [tutorial](https://www.python-graph-gallery.com/custom-fonts-in-matplotlib) I found, but it took me a while to figure out I had to nuke `fontlist-v330.json`. Anyhow,  given that dyslexia is not such an uncommon characteristic in an auditorium and that the word dyslexia doesn't look to appear in matplotlib documentation... I thought of, perhaps either:
- including an example using this particular font in this bit of documentation, or
- including a dyslexic-friendly font in matptlolib's standard font library.
The first will be easier I think, but the second may have more impact, though I have no clue how feasible that would be.
Any thoughts on this?

**Comment by saranti** [comment]
I'm going to send a PR for this so just to be clear, the work here is:
- Add a new "nuke/rebuild the font cache" function to the `matplotlib.font_manager` class.
- Write a new example using a dyslexic friendly font that shows how to add a new font file to `mpl-data/ttf/` and rebuild cache. (Maybe can go in https://matplotlib.org/stable/gallery/text_labels_and_annotations/font_file.html?)
- Maybe add a link to the "expose the font cache" function in the new example: https://matplotlib.org/stable/api/matplotlib_configuration_api.html#matplotlib.get_cachedir 
- Add a next_whats_new entry

**Comment by ernstki** [comment]
The documentation is [still lacking clear communication](https://matplotlib.org/stable/users/explain/text/fonts.html) about how to solve this issue.
